compiler/msgs.nim

```diff
@@ -734,20 +734,19 @@ proc outWriteln*(s: string) =
     flushFile(stdout)
 
 proc msgWriteln*(s: string) =
-  ## Writes to stdout. If --stdout option is given, writes to stderr instead.
+  ## Writes to stderr. If --stdout option is given, writes to stdout instead.
 
   #if gCmd == cmdIdeTools and optCDebug notin gGlobalOptions: return
 
   if not isNil(writelnHook):
     writelnHook(s)
   elif optStdout in gGlobalOptions:
-    if eStdErr in errorOutputs:
-      writeLine(stderr, s)
-      flushFile(stderr)
-  else:
     if eStdOut in errorOutputs:
       writeLine(stdout, s)
       flushFile(stdout)
+  else:
+    if eStdErr in errorOutputs:
+      writeLine(stderr, s)
```

Comment:
I don't know how you tested it, but before I added the flush calls it really was buffered and I could see the difference with my own eyes. *Maybe* stderr is unbuffered, but why rely on it? Please let's just keep these few calls, they cannot hurt us. You can put it in a 'when defined(windows)' if you really think every other OS follows Posix to the letter.